lib/config/applyrule.cpp

```diff
@@ -16,6 +16,12 @@ ApplyRule::ApplyRule(String targetType, String name, Expression::Ptr expression,
 	m_FVVar(std::move(fvvar)), m_FTerm(std::move(fterm)), m_IgnoreOnError(ignoreOnError), m_DebugInfo(std::move(di)), m_Scope(std::move(scope)), m_HasMatches(false)
 { }
 
+ApplyRule::ApplyRule(const ApplyRule& other)
+	: m_TargetType(other.m_TargetType), m_Name(other.m_Name), m_Expression(other.m_Expression), m_Filter(other.m_Filter),
+	m_Package(other.m_Package), m_FKVar(other.m_FKVar), m_FVVar(other.m_FVVar), m_FTerm(other.m_FTerm), m_IgnoreOnError(other.m_IgnoreOnError),
+	m_DebugInfo(other.m_DebugInfo), m_Scope(other.m_Scope), m_HasMatches(other.m_HasMatches.load(std::memory_order_relaxed))
+{ }
```

Comment:
Only needed for this line:
https://github.com/Icinga/icinga2/blob/3a8abdcc3b85a3be62a0805b465e5fac55ae0538/lib/config/applyrule.cpp#L78

This can also use `emplace_back`, which requires the constructor to be public, but I'd say that changing this is the lesser evil.

Reply:
> This can also use `emplace_back`, which requires the constructor to be public, but I'd say that changing this is the lesser evil.

Well, on second thought that won't work. Since the vector can relocate the objects, they have to be movable.

Reply:
I'd see if a move constructor would suffice. Another option would be to use pointers as value type for `m_Rules`, after all, `sizeof(ApplyRule)` is 256 (on my machine), so when the vector relocates, there's quite a bit to move. But that shouldn't make much of a difference performance-wise, but should keep the code more simple.